File: backend/src/ShaktiUdyog.Domain/Entities/Order.cs
using ShaktiUdyog.Domain.Constants;

namespace ShaktiUdyog.Domain.Entities;

/// <summary>
/// Confirmed customer order (requirements §14/§18). Customers see only
/// customer-visible fields and milestones; internal notes stay internal.
/// </summary>
public class Order
{
    public Guid Id { get; set; }
    public required string OrderNumber { get; set; }
    public Guid CompanyId { get; set; }
    public Company Company { get; set; } = null!;
    public Guid? QuotationId { get; set; }
    public Quotation? Quotation { get; set; }
    public string? PurchaseOrderReference { get; set; }
    public string Status { get; set; } = OrderStatuses.Confirmed;
    public DateTimeOffset PlacedAtUtc { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? PromisedDispatchDateUtc { get; set; }
    public string? DeliveryAddress { get; set; }
    public DateTimeOffset LastUpdatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    public List<OrderItem> Items { get; set; } = [];
    public List<OrderMilestone> Milestones { get; set; } = [];
    public List<Shipment> Shipments { get; set; } = [];
}

public class OrderItem
{
    public Guid Id { get; set; }
    public Guid OrderId { get; set; }
    public Order Order { get; set; } = null!;
    public required string PartNumber { get; set; }
    public required string Description { get; set; }
    public string? MaterialGrade { get; set; }
    public string? DrawingRevision { get; set; }
    public string Unit { get; set; } = "pcs";
    public int QuantityOrdered { get; set; }
    public int QuantityProduced { get; set; }
    public int QuantityDispatched { get; set; }
    /// <summary>Line value data is commercial; exposed to customers only in summaries.</summary>
    public decimal? UnitRate { get; set; }
}

public class OrderMilestone
{
    public Guid Id { get; set; }
    public Guid OrderId { get; set; }
    public Order Order { get; set; } = null!;
    public required string StatusCode { get; set; }
    /// <summary>Message shown to the customer when IsCustomerVisible is true.</summary>
    public string? CustomerMessage { get; set; }
    /// <summary>Internal production note — NEVER exposed through customer APIs.</summary>
    public string? InternalNote { get; set; }
    public bool IsCustomerVisible { get; set; } = true;
    public string ActorType { get; set; } = "Staff";
    public DateTimeOffset OccurredAtUtc { get; set; } = DateTimeOffset.UtcNow;
}

File: frontend/src/types/order.ts
/**
 * Order, Milestone, and Shipment domain models.
 */

import type { DocumentItem } from "./catalog";

export interface OrderItem {
  partNumber: string;
  description: string;
  materialGrade: string | null;
  drawingRevision: string | null;
  unit: string;
  quantityOrdered: number;
  quantityProduced: number;
  quantityDispatched: number;
}

export interface Shipment {
  id: string;
  transporter: string | null;
  trackingNumber: string | null;
  vehicleNumber: string | null;
  phoneNumber: string | null;
  dispatchDateUtc: string | null;
  estimatedArrivalUtc: string | null;
  deliveredAtUtc: string | null;
  hasProofOfDelivery: boolean;
}

export interface CreateShipmentPayload {
  transporter?: string | null;
  trackingNumber?: string | null;
  vehicleNumber?: string | null;
  phoneNumber?: string | null;
  dispatchDateUtc?: string | null;
  estimatedArrivalUtc?: string | null;
}

export interface TimelineEntry {
  statusCode: string;
  statusLabel: string;
  message: string | null;
  actorType: string;
  occurredAtUtc: string;
}

export interface OrderComment {
  authorRole: string;
  authorName: string | null;
  message: string;
  createdAtUtc: string;
}

export interface OrderListItem {
  id: string;
  orderNumber: string;
  status: string;
  statusLabel: string;
  placedAtUtc: string;
  promisedDispatchDateUtc: string | null;
  totalQuantity: number;
  lastUpdatedAtUtc: string;
  companyName: string | null;
  productType: string | null;
  assignedToUserId: string | null;
  assignedToName: string | null;
}

export interface OrderDetail {
  id: string;
  orderNumber: string;
  purchaseOrderReference: string | null;
  quotationId: string | null;
  status: string;
  statusLabel: string;
  statusDescription: string;
  placedAtUtc: string;
  promisedDispatchDateUtc: string | null;
  deliveryAddress: string | null;
  lastUpdatedAtUtc: string;
  items: OrderItem[];
  shipments: Shipment[];
  commercial: {
    invoiceNumber: string | null;
    invoiceDateUtc: string | null;
    dueDateUtc: string | null;
    total: number | null;
    amountPaid: number | null;
    balanceDue: number | null;
    paymentStatus: string | null;
  } | null;
  documents: DocumentItem[];
  assignedToUserId: string | null;
  assignedToName: string | null;
}

export interface EngineerOrder {
  id: string;
  orderNumber: string;
  companyName: string | null;
  productType: string | null;
  totalQuantity: number;
  manufacturingStage: string;
  stageUpdatedAt: string | null;
  placedAtUtc: string;
}
